### PR TITLE
#83 add generation element

### PR DESCRIPTION
**Pull request by armondhonore**

### CODE REVIEW

File: src/main/java/org/dasein/cloud/compute/VirtualMachineProduct.java [hunk 1]
```diff
@@ -39,6 +39,7 @@
     private Storage<Megabyte> ramSize;
     private float             standardHourlyRate;
     private VisibleScope      visibleScope;
+    private String			  generation;
```
Comment (clstokes) [inline src/main/java/org/dasein/cloud/compute/VirtualMachineProduct.java]: Let's change this to an enum called `status` with values `current` and `deprecated`.

Comment (armondhonore) [inline src/main/java/org/dasein/cloud/compute/VirtualMachineProduct.java]: Ok, I was thinking google might have other values, but if it is an enum we can always add to it.

Comment (clstokes) [inline src/main/java/org/dasein/cloud/compute/VirtualMachineProduct.java]: Even if Google has other values, we'll want to present a consistent _status_ across both.

Comment (armondhonore) [inline src/main/java/org/dasein/cloud/compute/VirtualMachineProduct.java]: Yes, I have already refactored this again to make it more abstract and require the provider implementation to deprecate the VM and if they don't have that concept it doesn't need to be implemented.

**Comment by clstokes** [comment]
@armondhonore can you fix the whitespace in these files, changing the tabs to 4-space indentions? This will make for a cleaner PR to the upstream project.

**Comment by armondhonore** [comment]
I will take a look at this I set it in my configs, but I don't think these are tabs I think this is another setting. I can look at setting this up so when I save a file it fixes it.

**Pull request merged**